Description: Make the Makefile use the $(DESTDIR) variable
Origin: vendor
Forwarded: not-needed
From: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-08-28

--- a/Makefile
+++ b/Makefile
@@ -104,7 +104,7 @@
 	$(CC) -MM *.c
 
 # Installation related variables and target
-PREFIX?=/usr/local
+PREFIX?=$(DESTDIR)/usr
 INCLUDE_PATH?=include/hiredis
 LIBRARY_PATH?=lib
 INSTALL_INCLUDE_PATH= $(PREFIX)/$(INCLUDE_PATH)
